prow/spyglass/lenses/junit/lens.go

```diff
@@ -120,13 +148,27 @@ func (lens Lens) Body(artifacts []lenses.Artifact, resourceDir string, data stri
 				for _, subSuite := range suite.Suites {
 					record(subSuite)
 				}
+
 				for _, test := range suite.Results {
-					result.junit = append(result.junit, test)
+					test := test
+					// There are cases where multiple entries of exactly the same
+					// testcase in a single junit result file, this could result
+					// from reruns of test cases by `go test --count=N` where N>1.
+					// Deduplicate them here in this case, and classify a test as being
+					// flaky if it both succeeded and failed
+					testFullName := fmt.Sprintf("%s...%s...%s", suite.Name, test.ClassName, test.Name)
+					if _, ok := groups[testFullName]; !ok {
+						groups[testFullName] = make([]junit.Result, 0)
```

Comment:
This shouldn't be necessary

```golang
var foo []string = nil
foo = append(foo, "hello")
```
works just fine

Reply:
interesting, seems like I've had the wrong impression for a while. done and thanks!